addons/stock/product.py

```diff
@@ -159,31 +177,232 @@ def _product_available(self, cr, uid, ids, field_names=None, arg=False, context=
 
         domain_move_in += domain_move_in_loc
         domain_move_out += domain_move_out_loc
-        moves_in = self.pool.get('stock.move').read_group(cr, uid, domain_move_in, ['product_id', 'product_qty'], ['product_id'], context=context)
-        moves_out = self.pool.get('stock.move').read_group(cr, uid, domain_move_out, ['product_id', 'product_qty'], ['product_id'], context=context)
-
         domain_quant += domain_quant_loc
-        quants = self.pool.get('stock.quant').read_group(cr, uid, domain_quant, ['product_id', 'qty'], ['product_id'], context=context)
-        quants = dict(map(lambda x: (x['product_id'][0], x['qty']), quants))
+        return domain_move_in, domain_move_out, domain_quant
+
+    def _product_available(self, cr, uid, ids, field_names=None, arg=False, context=None):
```

Comment:
Delete de sub method "def blazzing_fast_read_group" form _product_available method, and use the global method blazzing_fast_read_group.